src/Castle.Windsor.Tests/Compatibility/InterfaceMapping.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+
+#if FEATURE_NETCORE_REFLECTION_API
+namespace System.Reflection
+{
+	internal static class InterfaceMappingReflectionExtensions
+	{
+
+		public static InterfaceMapping GetInterfaceMap(this Type implType, Type interfaceType)
+		{
+			InterfaceMapping map;
+			map.TargetType = implType;
+			map.InterfaceType = interfaceType;
+
+			map.TargetMethods = implType.GetMethods().ToArray();
+			map.InterfaceMethods = new MethodInfo[map.TargetMethods.Length];
+			for(int i=0; i< map.TargetMethods.Length; i++)
+			{
+				map.InterfaceMethods[i] = GetCorrespondingMethodFromType(interfaceType, map.TargetMethods[i]);
+			}
+			return map;
+		}
+
+		// Given a MethodInfo which could be from any type, return a MethodInfo declared by the given type 
+		// that matches in name and parameter types.  Null is returned if there is no match.
+		//From https://github.com/dotnet/wcf/blob/master/src/System.Private.ServiceModel/src/System/ServiceModel/Description/TypeLoader.cs
```

Comment:
Just a reminder that Windsor is Apache 2 licensed, that code is MIT licensed so unless you are contributing it under the Apache 2 license as a Microsoft employee we can't take it.